### DISCUSSION TITLE
Modernize ntriples serializer

### DISCUSSION THREAD

**Discussion by ashleysommer** [RDFLib/rdflib#1580] [TO-DO list]
The NTriples serializer in rdflib is written in accordance with the RDF v1.0 2004 document here: https://www.w3.org/TR/2004/REC-rdf-testcases-20040210

In 2014 the NTriples v1.1 format specification was published: https://www.w3.org/TR/n-triples/

While there might have been a couple of tweaks made to the serializer since that release, it has never been fully brought into alignment with the NTriples v1.1 spec.

The one major difference to note is that v1.1 states all NTriples files are UTF-8 encoded, while our serializer forces all file outputs to be strictly ASCII encoded as-per the old 2004 spec.

**Comment by white-gecko** [comment]
This should be solved in combination with #1222 . Then everything is UTF-8 and only if a developer states anything different it will be different.
